### PR TITLE
Add generate fgs

### PR DESCRIPTION
**Pull request by Johannes-Sahlmann**
This PR introduces the source data and code necessary to generate the FGS SIAF. The comparison to FGS_SIAF_2019-04-15.xml shows a perfect match. 
 FGS_SIAF_2019-04-15.xml  was created to fix some minor bugs that affect the current PRD version of the file, see https://jira.stsci.edu/browse/JWSTSIAF-147.

### CODE REVIEW

**Comment by Johannes-Sahlmann** [comment]
@shanosborne can you please have a look at this PR and add the 'usual' comparisons etc. like you did for the other instruments. This time you can limit it to a comparison to M-024

**Comment by shanosborne** [comment]
@Johannes-Sahlmann I've added the difference txt files for M-024, FGS_SIAF_2018-04-17.xml, and FGS_SIAF_2019-04-15.xml

**Comment by Johannes-Sahlmann** [comment]
thank you for adding the new diff file @shanosborne !

**Comment by Johannes-Sahlmann** [comment]
I believe this is ready to be merged. With the latest changes, Colin's 2019-04-15 xml file is reproduced exactly, see below. The displayed differences are caused by the change in decimal precision between the old convention and the new.
[fgs_diff_FGS_SIAF_2019-04-15_pre_delivery.txt](https://github.com/spacetelescope/pysiaf/files/3753470/fgs_diff_FGS_SIAF_2019-04-15_pre_delivery.txt)

**Comment by Johannes-Sahlmann** [comment]
I think I messed up a little when I merged siaf-updates into this branch. @shanosborne we'll have to decide how to best clean this up.

**Comment by Johannes-Sahlmann** [comment]
Ready for review.
I think we should merge this now into siaf-updates and then create a new branch to accommodate the anticipated FGS changes.

File: generate/generate_reference_files.py [hunk 1]
```diff
@@ -38,13 +39,102 @@
...
 import pysiaf
-from pysiaf.constants import JWST_SOURCE_DATA_ROOT, JWST_PRD_VERSION
+from pysiaf.constants import JWST_SOURCE_DATA_ROOT, JWST_PRD_VERSION, JWST_DELIVERY_DATA_ROOT
 from pysiaf.utils import polynomial, tools
 from pysiaf import iando
 
 username = os.getlogin()
 timestamp = Time.now()
 
+def generate_fgs_fsw_coefficients(siaf=None, verbose=False):
```
Comment (shanosborne) [inline generate/generate_reference_files.py]: Found the missing docstring!

File: generate/generate_reference_files.py [hunk 2]
```diff
@@ -993,6 +1081,269 @@ def generate_siaf_pre_flight_reference_files_niriss(distortion_file_name, verbos
                                      delimiter_pad=' ', bookend=False)
 
 
+# FGS reference files
+def generate_siaf_pre_flight_reference_files_fgs(verbose=False, mode='siaf'):
+
```
Comment (shanosborne) [inline generate/generate_reference_files.py]: Another missing docstring here

Comment (Johannes-Sahlmann) [reply generate/generate_reference_files.py]: thanks, we can think about implementing pep8 speaks.

Comment (Johannes-Sahlmann) [reply generate/generate_reference_files.py]: added

File: generate/generate_reference_files.py (hunk 1 above)
Comment (Johannes-Sahlmann) [reply generate/generate_reference_files.py]: added

**Comment by Johannes-Sahlmann** [comment]
I will merge this because @ColinRCox indicated approval

**Pull request merged**